scripts/scr_planetary_feature/scr_planetary_feature.gml

```diff
@@ -1,7 +1,264 @@
+function planet_feature_bool(planet, search_feature){
+	var feature_count = array_length(planet);
+	var feature_exists = 0;
+	if (feature_count > 0){
+	for (var fc = 0; fc < feature_count; fc++;){
+		if (planet[fc].f_type == search_feature){
+			feature_exists = 1;
+		}
+		if (feature_exists == 1){break;}
+	}}
+	return feature_exists;	
+}
+
+
+//deletes all occurances of del_feature on planet
+function delete_features(planet, del_feature){
+	var delete_Array = search_planet_features(planet, del_feature)
+	if (array_length(delete_Array) >0){
+		for (var d=0;d<array_length(delete_Array);d++){
+			array_delete(planet, delete_Array[d],delete_Array[d]+1)
+		}
+	}
+}
+
+
+// returns 1 if an awake necron tomb iin system
+function awake_necron_Star(star){
+		for(var i = 1; i <= star.planets; i++){
+		if(awake_tomb_world(star.p_feature[i]) == 1)
+			{
+				return i;
+			}
+		}
+		return 0
+}
+
+
+//returns 1 if awake tomb world on planet 0 if tombs on planet but not awake and 2 if no tombs on planet
+function awake_tomb_world(planet){
+	var awake_tomb = 0;
+	 var tombs = search_planet_features(planet, P_features.Necron_Tomb);
+	 if (array_length(tombs)>0){
+		 for (var tomb =0;tomb<array_length(tombs);tomb++;){
+			 if (planet[tombs[tomb]].awake = 1){
+				awake_tomb = 1;
+			 }
+			 if (awake_tomb = 1){break;}
+		 }
+		 return awake_tomb;
+	 }
+	 return 2;
+}
+
+
+//selas a tomb world and switche off awake so will no longer spawn necrons or necron fleets
+function seal_tomb_world(planet){
+	var awake_tomb = 0;
+	 var tombs = search_planet_features(planet, search_feature);
+	 if (array_length(tombs)>0){
+		 for (var tomb =0;tomb<array_length(tombs);tomb++;){
+			 if (planet[tombs[tomb]].awake == 1){
+				awake_tomb = 1;
+				planet[tombs[tomb]].awake = 0;
+				planet[tombs[tomb]].sealed = 1;
+			 }
+			 if (awake_tomb = 1){break;}
+		 }
+	 }
+}
+
+
+//awakens a tomb world so necrons and necron fleets will spawn
+function awaken_tomb_world(planet){
+	var awake_tomb = 0;
+	 var tombs = search_planet_features(planet, search_feature);
+	 if (array_length(tombs)>0){
+		 for (var tomb =0;tomb<array_length(tombs);tomb++;){
+			 if (planet[tombs[tomb]].awake == 0){
+				awake_tomb = 1;
+				planet[tombs[tomb]].awake = 1;
+			 }
+			 if (awake_tomb = 1){break;}
+		 }
+		 
+	 }
+}
+
+
+// creates alerts for discovering features on a planet
+function scr_planetary_feature(planet_num) {
+	var plan_feat_count = array_length(p_feature[planet_num]);
+	//need to iterate over features instead of just looking at first
+	for (var f= 0; f < plan_feat_count;f++;){
+		var feat = p_feature[planet_num][f];
+		if (feat.player_hidden ==1){
+			feat.player_hidden =0;
+			switch (feat.f_type){
+				case P_features.Sororitas_Cathedral:
+					if (obj_controller.known[5]=0) then obj_controller.known[5]=1;
+				    var lop;lop="Sororitas Cathedral discovered on "+string(name)+" "+scr_roman(planet_num)+".";debugl(lop);
```

Comment:
also you can use a template string:
$"Sororitas Cathedral discovered on {string(name)} {scr_roman(planet_num)}."